modules/ROOT/pages/Development/UpdatingFromSml35.adoc

```diff
@@ -121,6 +182,12 @@ C3861	'DOREPLIFETIME': identifier not found
 C2275 'TheClassYouHadAsTheFirstArgument': expected an expression instead of a type
 ```
 
+=== Enhanced Input System {cpp} Changes
+
+SML no longer provides the interface for creating gameplay tag->input action mappings, you must instead make a custom SomethingModule in {cpp} containing a map of gameplay tag->input action, then retrieve data from that instead.
```

Comment:
The GameInstanceModule no longer provides the interface for creating GameplayTag->InputAction mappings, since it was previously modifying one of the game's config classes, which is unsafe.

Previously, to bind to an action using a GameplayTag, you would add your mappings to the GameInstanceModule blueprint and use the following in {cpp}:

```cpp
EnhancedInputComponent->BindActionByTag(FGameplayTag::RequestGameplayTag(TEXT("Tag.Name")), ETriggerEvent::Triggered, this, &Class::Function);
```

Instead, you should now create a custom GameInstanceModule class in {cpp} containing a `TMap<FGameplayTag, UInputAction*>`, and use your class as the parent of your GameInstanceModule blueprint:

```cpp
// Includes left out

UCLASS(Blueprintable)
class MYMOD_API UMyModGameInstanceModule : public UGameInstanceModule {
    GENERATED_BODY()
public:
    UPROPERTY(EditDefaultsOnly, Category = "Advanced | Input")
    TMap<UInputAction*, FGameplayTag> InputActionTagBindings;
};
```

Then, to use the bindings, you would do as follows:

```cpp
UGameInstance* GameInstance = GetWorld()->GetGameInstance(); // May vary depending on the class you're using this from 
UGameInstanceModuleManager* GameInstanceModuleManager = GameInstance->GetSubsystem<UGameInstanceModuleManager>();
UMyModGameInstanceModule* MyModGameInstanceModule = Cast<UMyModGameInstanceModule>(GameInstanceModuleManager ->FindModule(TEXT("MyMod")));
EnhancedInputComponent->BindAction(MyModGameInstanceModule->InputActionTagBindings[FGameplayTag::RequestGameplayTag(TEXT("Tag.Name"))], ETriggerEvent::Triggered, this, &Class::Function);
```

Reply:
Could not add this as a suggestion because of the ```

Also this template was not tested, as I have no mod to test it on